observationtree/src/main/scala/explore/observationtree/TargetTree.scala

```diff
@@ -0,0 +1,181 @@
+        val targetObsIds = obsByTarget.get(target).fold(List.empty[String])(_.map(_.id.toString))
+
+        target.name -> AtlasTree.Item[TreeItem](
+          target.name,
+          targetObsIds,
+          hasChildren = targetObsIds.nonEmpty,
+          isExpanded = !collapsedTargetIds.contains(target.name),
+          isChildrenLoading = false,
+          TargetItem(target, targetObsIds.length)
+        )
+      }
+
+      val obsData = observations.map(obs =>
+        obs.id.toString -> AtlasTree.Item[TreeItem](
+          obs.id.toString,
+          List.empty,
+          hasChildren = false,
+          isExpanded = false,
+          isChildrenLoading = false,
+          ObsItem(obs)
+        )
+      )
+
+      AtlasTree.Data(
+        rootId = "",
+        items = ((root +: targetData) ++ obsData).toMap
+      )
+    }
+
+    protected def renderItem(params: AtlasTree.RenderItemParams[TreeItem]): VdomNode = {
+      def wrap(divAttrs: TagMod*)(element: VdomNode, overlay: VdomNode): VdomNode =
+        <.div(
+          <.div(^.position.absolute, ^.zIndex := "2", ^.pointerEvents.none)(divAttrs: _*)(overlay),
+          <.div(divAttrs: _*)(element)
+        )
+
+      def handleClick: Callback =
+        params.item.isExpanded
+          .forall(identity)
+          .fold(params.onCollapse(params.item.id), params.onExpand(params.item.id))
+
+      def opIcon(hasChildren: Boolean) =
+        hasChildren.fold(
+          Icon(
+            "chevron " + params.item.isExpanded.forall(identity).fold("down", "right")
+          )(^.cursor.pointer, ^.onClick --> handleClick),
+          Icon("chevron right")
+        )
+
+      def dragIcon =
+        <.span(^.float.right, ^.pointerEvents.all, params.provided.dragHandleProps, Icon("sort"))
+
+      <.div(params.provided.innerRef, params.provided.draggableProps)(
+        <.div(^.margin := "5px")(
+          params.item.data
+            .map[VdomNode] {
+              _ match {
+                case TargetItem(target, obsCount) =>
+                  <.span(
+                    opIcon(obsCount > 0),
+                    target.name,
+                    <.span( /*^.display.none,*/ params.provided.dragHandleProps, "*"),
+                    <.span(^.float.right, s"$obsCount Obs")
+                  )
+                case ObsItem(obs)                 =>
+                  wrap(^.width := "225px")(
+                    ObsBadge(obs, ObsBadge.Layout.ConfAndConstraints),
+                    dragIcon
+                  )
+              }
+            }
+        )
+      )
+    }
+
+    val onCollapse =
+      (itemId: AtlasTree.ItemId, _: AtlasTree.Path) =>
+        $.modStateL(State.collapsedTargetIds)(_ + itemId)
+
+    val onExpand =
+      (itemId: AtlasTree.ItemId, _: AtlasTree.Path) =>
+        $.modStateL(State.collapsedTargetIds)(_ - itemId)
+
+    def onDragEnd(source: AtlasTree.Position, destination: Option[AtlasTree.Position]): Callback =
+      Callback(
```

Comment:
`Callback.log`?

Reply:
Good gatch. Anyway, this should go away, it's debugging code.